packages/trees/backend/sirius-components-trees/src/test/java/org/eclipse/sirius/components/trees/renderer/TreeRendererTests.java

```diff
@@ -0,0 +1,249 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.trees.renderer;
+
+import static java.util.stream.Collectors.joining;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import org.eclipse.sirius.components.core.api.labels.StyledString;
+import org.eclipse.sirius.components.core.api.labels.StyledStringFragment;
+import org.eclipse.sirius.components.representations.Success;
+import org.eclipse.sirius.components.representations.VariableManager;
+import org.eclipse.sirius.components.trees.Tree;
+import org.eclipse.sirius.components.trees.TreeItem;
+import org.eclipse.sirius.components.trees.description.TreeDescription;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test class for the {@link TreeRenderer}.
+ *
+ * @author Arthur Daussy
+ */
+public class TreeRendererTests {
```

Comment:
Please prefer an integration test to validate the whole process

Reply:
Please don't make me do that :cry:

Unit testing is life...

It this particular case it allowed me to easily create the "StackOverflow" use case and use a TDD approach. It even help me to fix to issue. It help me to detect the need of the index to give the capacity for the specifier to create truly unique id.

 Could you explain why you prefer integration test of unit test please?